### PR TITLE
#1018 fix. Scrab chase logic less janky

### PR DESCRIPTION
**Pull request by d3nwah**
https://github.com/AliveTeam/alive_reversing/issues/1018

### CODE REVIEW

File: Source/AliveLibAE/Scrab.cpp [hunk 1]
```diff
@@ -1634,52 +1614,50 @@ __int16 Scrab::AI_ChasingEnemy_State_Running_2(BaseAliveGameObject* pObj)
...
-            {
-                field_108_next_motion = eScrabMotions::M_Turn_3_4A91A0;
-            }
-            else
+            if (field_106_current_motion != eScrabMotions::M_JumpToFall_8_4A9220)
             {
-                field_106_current_motion = eScrabMotions::M_Turn_3_4A91A0;
-                field_108_next_motion = -1;
-                MapFollowMe_408D10(TRUE);
+                //A patch workaround
```
Comment (paulsapps) [inline Source/AliveLibAE/Scrab.cpp:1638]: what does this mean?

File: Source/AliveLibAE/Scrab.cpp [hunk 2]
```diff
@@ -1246,15 +1246,18 @@ __int16 Scrab::AI_ChasingEnemy_1_4A6470()
...
         return AI_ChasingEnemy::eState1_Panic_4;
     }
 
-    LiftPoint* pLiftPoint = static_cast<LiftPoint*>(pObj);
-    if (pLiftPoint && pLiftPoint->field_4_typeId == Types::eLiftPoint_78 && 
-        !pLiftPoint->vOnAnyFloor_461920() && field_11C_sub_state != AI_ChasingEnemy::eState1_Panic_4)
+    LiftPoint* pLiftPoint = static_cast<LiftPoint*>(sObjectIds_5C1B70.Find_449CF0(field_110_id));
+    if (!pLiftPoint || pLiftPoint->field_4_typeId != Types::eLiftPoint_78)
+    {
+        pLiftPoint = nullptr;
```
Comment (paulsapps) [inline Source/AliveLibAE/Scrab.cpp:1258]: comment that this is an OG fix for pLiftPoint being used later even if it isn't a lift point

File: Source/AliveLibAE/Scrab.cpp (hunk 1 above)
Comment (d3nwah) [reply Source/AliveLibAE/Scrab.cpp:1638]: It's just that this scenario should probably never happen in the first place, the condition is like a plaster on it.

Comment (d3nwah) [reply Source/AliveLibAE/Scrab.cpp:1638]: I forgot to finish that comment :tapeDan:

Comment (paulsapps) [reply Source/AliveLibAE/Scrab.cpp:1638]: maybe reword it cause I had no idea what it meant ⚰️
